File: apps/silo_app/lib/src/protocol/sandbox.dart
/// Mirrors `silo_core::sandbox::AccessReport`.
library;

class AccessReport {
  const AccessReport({
    required this.sandboxKind,
    required this.workspaceMount,
    required this.scratchDir,
    required this.readablePaths,
    required this.allowedDomains,
    required this.credentialDomains,
    required this.notes,
  });

  final String sandboxKind;
  final String workspaceMount;
  final String scratchDir;
  final List<String> readablePaths;
  final List<String> allowedDomains;
  final List<String> credentialDomains;
  final List<String> notes;

  factory AccessReport.fromJson(Map<String, dynamic> json) => AccessReport(
        sandboxKind: json['sandbox_kind'] as String,
        workspaceMount: json['workspace_mount'] as String,
        scratchDir: json['scratch_dir'] as String,
        readablePaths: _stringList(json['readable_paths']),
        allowedDomains: _stringList(json['allowed_domains']),
        credentialDomains: _stringList(json['credential_domains']),
        notes: _stringList(json['notes']),
      );

  Map<String, dynamic> toJson() => {
        'sandbox_kind': sandboxKind,
        'workspace_mount': workspaceMount,
        'scratch_dir': scratchDir,
        'readable_paths': readablePaths,
        'allowed_domains': allowedDomains,
        'credential_domains': credentialDomains,
        'notes': notes,
      };
}

List<String> _stringList(Object? value) =>
    (value as List<dynamic>? ?? const []).cast<String>();

File: crates/silo-core/src/sandbox.rs
//! Types describing what a sandbox exposes to the LLM and to the user.

use serde::{Deserialize, Serialize};

/// Human-readable description of everything the sandboxed LLM can reach.
/// Surfaced to interactive clients so the user can audit access at a glance.
#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
pub struct AccessReport {
    /// Sandbox backend in use, e.g. "mock" or "macos-sandbox-exec".
    pub sandbox_kind: String,
    /// Path of the read/write workspace as seen inside the sandbox.
    pub workspace_mount: String,
    /// Path of the per-sandbox writable scratch space inside the sandbox.
    pub scratch_dir: String,
    /// Host paths the sandbox can read (and execute) but not write.
    pub readable_paths: Vec<String>,
    /// Domains the egress proxy will allow.
    pub allowed_domains: Vec<String>,
    /// Domains for which the proxy injects credentials. Only domain names
    /// are listed; the credentials themselves are never exposed.
    pub credential_domains: Vec<String>,
    /// Free-form caveats, e.g. platform-specific lock limitations.
    pub notes: Vec<String>,
}
